src/UnitTests/GitHub.VisualStudio/UI/Views/GitHubPaneViewModelTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿using GitHub.Api;
+using GitHub.Exports;
+using GitHub.Models;
+using GitHub.Primitives;
+using GitHub.Services;
+using GitHub.UI;
+using GitHub.VisualStudio;
+using GitHub.VisualStudio.UI.Views;
+using NSubstitute;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.ComponentModel.Design;
+using System.Linq;
+using System.Reactive.Linq;
+using UnitTests;
+using Xunit;
+
+public class GitHubPaneViewModelTests : TestBaseClass
+{
+    private readonly IServiceProvider serviceProvider;
+    private readonly IUIController uiController;
+    private readonly FakeMenuCommandService menuCommandService;
+    private readonly GitHubPaneViewModel viewModel;
+    private UIViewType lastUiControllerJump;
+
+    public GitHubPaneViewModelTests()
+    {
+        #region Start tests already logged in with an active repo.
+
+        var repositoryHosts = Substitutes.RepositoryHosts;
+        repositoryHosts.IsLoggedInToAnyHost.Returns(true);
+
+        var teamExplorerServiceHolder = Substitute.For<ITeamExplorerServiceHolder>();
+        var activeRepo = Substitute.For<ILocalRepositoryModel>();
+        activeRepo.CloneUrl.Returns(new UriString("https://github.com/foo/foo"));
+        teamExplorerServiceHolder
+            .When(_ => _.Subscribe(Arg.Any<object>(), Arg.Any<Action<ILocalRepositoryModel>>()))
+            .Do(_ =>
+            {
+                var invokeAction = _.Arg<Action<ILocalRepositoryModel>>();
+                invokeAction(activeRepo);
+            });
+
+        var connectionManager = Substitutes.ConnectionManager;
+        var connection = Substitutes.Connection;
+        var connectionHostAddress = HostAddress.Create(activeRepo.CloneUrl.ToString());
+        connection.HostAddress.Returns(connectionHostAddress);
+        connectionManager.Connections.Returns(new ObservableCollection<IConnection>(new[] {
+                connection
+            }));
+        connection.Login().Returns(Observable.Return(connection));
+
+        var host = Substitute.For<IRepositoryHost>();
+        host.IsLoggedIn.Returns(true);
+        repositoryHosts.LookupHost(connectionHostAddress).Returns(host);
+
+        #endregion
+
+        #region Wire up service/UI provider.
+
+        serviceProvider = Substitutes.GetFullyMockedServiceProvider();
+        menuCommandService = new FakeMenuCommandService();
+        serviceProvider.GetService(typeof(IMenuCommandService)).Returns(menuCommandService);
+
+        // TODO: Maybe could avoid cast by making a new interface that inherits IServiceProvider and IUIProvider
```

Comment:
Yeah, after re-reading that comment, it definitely didn't make sense; removed it.

It felt weird to cast the mock IServiceProvider to IUIProvider. It does work because of the way the mock is set up, but the consumer wouldn't know that based solely on the return type of GetFullyMockedServiceProvider. Maybe Substitutes could have a method that returns IUIProvider?